src/statistics.jl

```diff
@@ -0,0 +1,88 @@
+abstract type AbstractStatistic end
+
+struct Mean{D<:AbstractDomain} <: AbstractStatistic
+  domain::D
+  values::Dict{Symbol, Vector} 
+end
+
+function Mean(solution::SimulationSolution)
+  # output dictionary
+  values = Dict(var => mean(real) for (var, real) in solution.realizations)
+    
+  Mean{typeof(solution.domain)}(solution.domain, values)
+end
+
+mean(solution::SimulationSolution) = Mean(solution)
+
+struct Variance{D<:AbstractDomain} <: AbstractStatistic
+  domain::D
+  values::Dict{Symbol, Vector} 
+end
+
+function Variance(solution::SimulationSolution)
+  # output dictionary
+  values = Dict(var => var(real) for (var, real) in solution.realizations)
+    
+  Variance{typeof(solution.domain)}(solution.domain, values)
+end
+
+var(solution::SimulationSolution) = Variance(solution)
+
+struct Quantile{D<:AbstractDomain, P<:Real}
+  domain::D
+  p::P
+  values::Dict{Symbol, Vector}
+end
+
+function Quantile(solution::SimulationSolution, p::Real)
+  # function returning vector of slices
+  slices(x) = [getindex.(x, i) for i in 1:npoints(solution.domain)]
+  
+  # output dictionary
+  values = Dict(var => quantile.(slices(real), p) for (var, real) in solution.realizations)
```

Comment:
I think we should be clearer here, the code is not easy to read. Maybe we should do an explicit loop over `solution.realizations` and for each entry perform the quantile calculation. I can work on this after you write some tests. :+1: